src/Infrastructure/Riganti.Utils.Infrastructure.EntityFramework/EntityFrameworkUnitOfWork.cs

```diff
@@ -90,28 +85,56 @@ protected EntityFrameworkUnitOfWork(IUnitOfWorkProvider unitOfWorkProvider, Func
         }
 
         /// <summary>
-        /// Commits this instance when we have to.
+        /// Gets the <see cref="DbContext" />.
+        /// </summary>
+        public TDbContext Context { get; }
+
+        /// <inheritdoc cref="ICheckChildCommitUnitOfWork.Parent"/>
+        public IUnitOfWork Parent { get; }
+
+        /// <inheritdoc cref="ICheckChildCommitUnitOfWork.CommitRequested"/>
+        public bool CommitRequested { get; private set; }
+
+        /// <summary>
+        /// Commits this instance when we have to. Skip and request from parent, if we don't own the context.
         /// </summary>
         public override void Commit()
         {
             if (HasOwnContext())
             {
+                CommitRequested = false;
                 base.Commit();
             }
+            else
+            {
+                TryRequestParentCommit();
+            }
         }
-        
+
         /// <summary>
-        /// Commits this instance when we have to.
+        /// Commits this instance when we have to. Skip and request from parent, if we don't own the context.
         /// </summary>
         public override Task CommitAsync()
         {
             if (HasOwnContext())
             {
+                CommitRequested = false;
```

Comment:
I would suggest to rename `CommitRequested` to something like `CommitPending`, because in this case it is not true, that the commit was not requested. It was requested but also was satisfied :) So it is not pending.